packages/backend/src/ee/services/AiAgentService.ts

```diff
@@ -769,6 +883,23 @@ export class AiAgentService {
             throw new ForbiddenError();
         }
 
+        // Check group access for the agent
+        if (thread.agentUuid) {
+            const agent = await this.aiAgentModel.getAgent({
+                organizationUuid,
+                agentUuid: thread.agentUuid,
+            });
+
+            if (agent) {
+                const hasAccess = await this.checkAgentGroupAccess(user, agent);
+                if (!hasAccess) {
+                    throw new ForbiddenError(
+                        'Insufficient permissions to stream response for this agent',
+                    );
+                }
+            }
```

Comment:
`checkAgentThreadAccess`?

Reply:
done

Reply:
focussing on this now - the thing is that ai_thread does not have info from the user, so have to do an extra query to check user uuid from a web app thread for example